### ISSUE TITLE
Ability to delete an ADObject

### ISSUE THREAD

**Issue by ammarisrar** [zorn96/ms_active_directory#45]
Hello,

I have been going through the documentation, but I couldn't find any snippets to delete the ADObject. Is there some way to achieve this with the current implementation, or is this feature not added yet?

**Comment by mrgum** [comment]
+1 

Looking at session and a user object (using inspect.getmembers(self.session, predicate=inspect.ismethod)) I do not see any remove methods...

session object
```
ADSession.__init__
ADSession.__repr__
ADSession.__str__
ADSession._create_object
ADSession._do_something_to_attribute_for_object
ADSession._figure_out_search_attributes_for_user_or_group
ADSession._find_ad_objects_and_attrs
ADSession._find_by_name_common
ADSession._something_members_to_or_from_groups
ADSession._validate_computer_and_get_computer_obj
ADSession._validate_group_and_get_group_obj
ADSession._validate_obj_and_get_ad_obj
ADSession._validate_user_and_get_user_obj
ADSession.add_computers_to_groups
ADSession.add_groups_to_groups
ADSession.add_permission_to_computer_security_descriptor
ADSession.add_permission_to_group_security_descriptor
ADSession.add_permission_to_object_security_descriptor
ADSession.add_permission_to_user_security_descriptor
ADSession.add_users_to_groups
ADSession.atomic_append_to_attribute_for_computer
ADSession.atomic_append_to_attribute_for_group
ADSession.atomic_append_to_attribute_for_object
ADSession.atomic_append_to_attribute_for_user
ADSession.atomic_append_to_attributes_for_computer
ADSession.atomic_append_to_attributes_for_group
ADSession.atomic_append_to_attributes_for_object
ADSession.atomic_append_to_attributes_for_user
ADSession.change_password_for_account
ADSession.create_computer
ADSession.create_group
ADSession.create_transfer_sessions_to_all_trusted_domains
ADSession.create_unmanaged_user
ADSession.create_user
ADSession.disable_account
ADSession.dn_exists_in_domain
ADSession.enable_account
ADSession.find_certificate_authorities_for_domain
ADSession.find_computer_by_distinguished_name
ADSession.find_computer_by_name
ADSession.find_computer_by_principal_name
ADSession.find_computer_by_sam_name
ADSession.find_computer_by_sid
ADSession.find_computers_by_attribute
ADSession.find_computers_by_common_name
ADSession.find_current_time_for_domain
ADSession.find_dns_servers_for_domain
ADSession.find_forest_schema_version
ADSession.find_functional_level_for_domain
ADSession.find_group_by_distinguished_name
ADSession.find_group_by_name
ADSession.find_group_by_sam_name
ADSession.find_group_by_sid
ADSession.find_groups_by_attribute
ADSession.find_groups_by_common_name
ADSession.find_groups_for_computer
ADSession.find_groups_for_computers
ADSession.find_groups_for_entities
ADSession.find_groups_for_group
ADSession.find_groups_for_groups
ADSession.find_groups_for_user
ADSession.find_groups_for_users
ADSession.find_members_of_group
ADSession.find_members_of_group_recursive
ADSession.find_members_of_groups
ADSession.find_members_of_groups_recursive
ADSession.find_netbios_name_for_domain
ADSession.find_object_by_canonical_name
ADSession.find_object_by_distinguished_name
ADSession.find_object_by_sid
ADSession.find_objects_with_attribute
ADSession.find_policies_in_domain
ADSession.find_primary_group_for_account
ADSession.find_security_descriptor_for_computer
ADSession.find_security_descriptor_for_group
ADSession.find_security_descriptor_for_object
ADSession.find_security_descriptor_for_user
ADSession.find_sid_for_domain
ADSession.find_supported_sasl_mechanisms_for_domain
ADSession.find_trusted_domains_for_domain
ADSession.find_upn_suffixes_for_domain
ADSession.find_user_by_distinguished_name
ADSession.find_user_by_name
ADSession.find_user_by_principal_name
ADSession.find_user_by_sam_name
ADSession.find_user_by_sid
ADSession.find_users_by_attribute
ADSession.find_users_by_common_name
ADSession.get_current_server_uri
ADSession.get_domain
ADSession.get_domain_dns_name
ADSession.get_domain_search_base
ADSession.get_ldap_connection
ADSession.get_search_paging_size
ADSession.get_trusted_domain_cache_lifetime_seconds
ADSession.is_anonymous
ADSession.is_authenticated
ADSession.is_domain_close_in_time_to_localhost
ADSession.is_encrypted
ADSession.is_open
ADSession.is_session_user_from_domain
ADSession.is_thread_safe
ADSession.object_exists_in_domain_with_attribute
ADSession.overwrite_attribute_for_computer
ADSession.overwrite_attribute_for_group
ADSession.overwrite_attribute_for_object
ADSession.overwrite_attribute_for_user
ADSession.overwrite_attributes_for_computer
ADSession.overwrite_attributes_for_group
ADSession.overwrite_attributes_for_object
ADSession.overwrite_attributes_for_user
ADSession.remove_computers_from_groups
ADSession.remove_groups_from_groups
ADSession.remove_users_from_groups
ADSession.reset_password_for_account
ADSession.set_computer_security_descriptor
ADSession.set_domain_search_base
ADSession.set_group_security_descriptor
ADSession.set_object_security_descriptor
```
a user object
```
ADSession.set_primary_group_for_account
ADSession.set_search_paging_size
ADSession.set_trusted_domain_cache_lifetime_seconds
ADSession.set_user_security_descriptor
ADSession.take_over_existing_computer
ADSession.unlock_account
ADSession.validate_location_for_creation_or_movement_and_get_dn
ADSession.who_am_i
ADObject.__eq__
ADObject.__hash__
ADUser.__init__
ADObject.__repr__
ADObject.__str__
ADObject.get
ADObject.has_policy_directly_enabled
ADObject.is_of_object_class
```

**Comment by mrgum** [comment]
I've implemented this for users and groups in #54 is there something else you want deleted?
